### PR TITLE
SNOW-1346233 execute automated external browser tests

### PR DESCRIPTION
**Pull request by sfc-gh-akolodziejczyk**
### Description
[SNOW-1346233](https://snowflakecomputing.atlassian.net/browse/SNOW-1346233)
1. Authentication tests will be stored in a separate directory `test/authentication`
2. Credentials are stored in separate gpg file.
3. Authentication tests will be executed on Jenkins in parallel to other tests.
4. Authentication tests are executed inside the docker container which is created from images available in the internal repository. It includes code responsible for filling in browser credentials. Code for building the image will be stored in a separate repository

### Checklist
- [ ] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [ ] Create tests which fail without the change (if possible)
- [ ] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [ ] Extend the README / documentation and ensure is properly displayed (if necessary)
- [ ] Provide JIRA issue id (if possible) or GitHub issue id in commit message

[SNOW-1346233]: https://snowflakecomputing.atlassian.net/browse/SNOW-1346233?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by sfc-gh-dprzybysz — APPROVED** [review]

File: test/authentication/testExternalBrowser.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  }
+});
+
+async function cleanBrowserProcesses() {
+  exec('pkill -f chromium');
+  exec('pkill -f xdg-open');
+}
+
+async function verifyConnectionIsUp(connection) {
+  assert.ok(connection.isUp(), 'Connection is not up');
```
Comment (sfc-gh-pmotacki) [inline test/authentication/testExternalBrowser.js:136]: We can skip verifying whether the connection is up. This is controlled inside the isValidAsync method.

File: test/authentication/testExternalBrowser.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  assert.ok(await connection.isValidAsync(), 'Connection is not valid');
+  await testUtil.executeCmdAsync(connection, 'Select 1');
+}
+
+async function verifyConnectionIsNotUp(connection, message = 'Unable to perform operation because a connection was never established.') {
+  assert.ok(!(connection.isUp()), 'Connection should not be up');
```
Comment (sfc-gh-pmotacki) [inline test/authentication/testExternalBrowser.js:142]: The same as in a previous comment

File: test/integration/testManualConnection.js [hunk 3]
```diff
@@ -201,36 +64,6 @@ if (process.env.RUN_MANUAL_TESTS_ONLY === 'true') {
       });
     });
 
-    describe('Connection test - okta', function () {
```
Comment (sfc-gh-pmotacki) [inline test/integration/testManualConnection.js:204]: Is this test replaced by the new one?

Comment (sfc-gh-akolodziejczyk) [reply test/integration/testManualConnection.js]: it will be in next pr, restored for now

File: test/authentication/testExternalBrowser.js (hunk 1 above)
Comment (sfc-gh-akolodziejczyk) [reply test/authentication/testExternalBrowser.js]: done

**Review by sfc-gh-pmotacki — APPROVED** [review]

**Pull request merged**